### PR TITLE
Homework Week 1

### PR DESCRIPTION
**Pull request by Swapna-Sahu**

### CODE REVIEW

File: week1/HomeworkWeek1.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# Task 1 - Write a python script to print your name and age
+
+name = input("Enter your name")
+age = int(input("Enter your age"))
+
+print("Your name is : ", name, " Your age is : ", age)
```
Comment (Rassibassi) [inline week1/HomeworkWeek1.py]: You remember that in JS you can populate a string with variables using template strings as shown here:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

The same you can do with Python, but it is called f-strings:
```
name = "Eric"
age = 74
print(f"Hello, {name}. You are {age}.")
```

File: week1/HomeworkWeek1.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+numbers = {0: 10, 1: 20}
+numbers[2] = 30
+print(numbers)
+
+# Task 5 - Write a Python program to calculate body mass index.
+name = input("enter your name")
+weight = float(input("enter your weight in kg"))
+height = float(input("enter your height in meter"))
+
+bmi = weight/height*height
```
Comment (Rassibassi) [inline week1/HomeworkWeek1.py]: In Python there is a shortcut to do a squared calculation, `a*a` is the same as `a**2` where the `**` stands for `to the power of. So `a**2` is `a to the power of 2`

File: week1/HomeworkWeek1.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+print(name, "'s BMI is ", bmi)
+
+# Task 6 - Guess a number game - between 1 to 9.
+
+guessnumber = 7
```
Comment (Rassibassi) [inline week1/HomeworkWeek1.py]: In all your code you use `snake_case`. Be consistent and please also do `guess_number` :) same for `user_guess`

File: week1/HomeworkWeek1.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+while True:
+    userguess = int(input("Enter your guess number"))
+    if userguess == guessnumber:
+        print("Well guessed!")
+        break
+
+# Task 7 - Create a tuple with different data types
+
+tuple1 = ("hello world", 134, True, 40.4)
```
Comment (Rassibassi) [inline week1/HomeworkWeek1.py]: It's a silly example exercise, but using a variable name like `tuple1` which is very close to `tuple` (a word from pythons syntax) is not a good choice. What about `example_tuple` or `sample_tuple`.

File: week1/HomeworkWeek1.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+print(tuple1)
+
+# Task 8 - Create a list of 5 city names and convert it into tuples.
+
+city = ["Delhi", "Mumbai", "Kolkata", "Chennai", "Copenhagen"]
+print("Type of city variable : ", type(city))
+citytuple = tuple(city)
```
Comment (Rassibassi) [inline week1/HomeworkWeek1.py]: `city_tuple` if we were consistent :)

File: week1/HomeworkWeek1.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+print("Type of citytuple variable : ", type(citytuple))
+print("City Tuple : ", citytuple)
+
+# Task 9 - Remove duplicated from the list
+a = [10, 20, 30, 20, 10, 50, 60, 40, 80, 50, 40]
+a = list(dict.fromkeys(a))
+print(a)
+
```
Comment (Rassibassi) [inline week1/HomeworkWeek1.py:66]: You are using the property of a Python dictionary, that its keys are unique. It works and is kind of elegant, but also not the purpose of a dictionary.

Google what a `set` is and try to use a set instead :)

Further, I know that the exercise uses the variable name `a`, as above, I would rename that to `sample_list`.

File: week1/HomeworkWeek1.py [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+str2 = ''
+
+for i in range(len(str1)):
+    if (i % 2 == 0):
+        str2 = str2 + str1[i]
+
+print("Original String :  ", str1)
+print("Final String :     ", str2)
+
```
Comment (Rassibassi) [inline week1/HomeworkWeek1.py:82]: Very good. This example works :)

Now my feedback:

First, very bad choice of variable names here :(

Next, Python can do this more elegant or for a _Pythonista_, there is a more _pythonic_ way of doing this.

In Python **a string is basically a list of characters**, or in other words, in Python like a list also a string is **iterable**,
and a list can be looped through like this:
```python
sample_list = [100,200,300]
for sample_int in sample_list:
    print(sample_int)
```
Do you see that there is no indexing `sample_list[i]` necessary anymore? You can do the same with a string

Now, you might have also seen that we lost the index number `i` within the loop. Yet, there is also a pythonic way of getting a running number into your for loop, see here:
```python
sample_list = [100,200,300]
for i, sample_int in enumerate(sample_list):
    print(i, sample_int)
```

Can you use my feedback and improve your code?

A last little hint, these two lines do the same thing :) but the second line is more elegant
```python
a = a + b
a += b
```

**Review by Rassibassi — COMMENTED** [review]
Good job overall! And also for doing all the extra exercises :)

See my inline comments for some improvements :)

File: week1/HomeworkWeek1.py (hunk 1 above)
Comment (Swapna-Sahu) [reply week1/HomeworkWeek1.py]: Ah ok . Thanks. I will do.

File: week1/HomeworkWeek1.py (hunk 2 above)
Comment (Swapna-Sahu) [reply week1/HomeworkWeek1.py]: For sure. Thanks.

File: week1/HomeworkWeek1.py (hunk 3 above)
Comment (Swapna-Sahu) [reply week1/HomeworkWeek1.py]: Agree with you. My mistake. Will change it.

File: week1/HomeworkWeek1.py (hunk 4 above)
Comment (Swapna-Sahu) [reply week1/HomeworkWeek1.py]: done.

File: week1/HomeworkWeek1.py (hunk 5 above)
Comment (Swapna-Sahu) [reply week1/HomeworkWeek1.py]: Sure.

File: week1/HomeworkWeek1.py (hunk 6 above)
Comment (Swapna-Sahu) [reply week1/HomeworkWeek1.py:66]: Ok. Set is also working but it is not in order way.  I think it is checking value and removing the duplicate not the removing from index. I did using set also. 
Thanks for letting me.

File: week1/HomeworkWeek1.py (hunk 7 above)
Comment (Swapna-Sahu) [reply week1/HomeworkWeek1.py:82]: Thanks for explaining. I have updated my file with ur feedback.

**Pull request merged**